src/commands/create/index.ts

```diff
@@ -65,181 +104,143 @@ export default class Create extends Command {
-            return languageEntry;
+        Create.flags.core.options = packageIndex.getCores().map((core: Core) => {
+            return core.id;
         });
-        Create.flags.interface.options = availableInterfaces.map((interfaceEntry: FunctionalInterfaceDescription) => {
-            return interfaceEntry.value;
+        Create.flags.interface.options = packageIndex.getExtensions().map((ext: Extension) => {
+            return ext.id;
         });
     }
 
-    private _getScriptExecutionPath(sdkConfig: SdkConfig): string {
+    private _getScriptExecutionPath(coreConfig: CorePackageConfig): string {
         const basePath = process.env.VELOCITAS_SDK_PATH_OVERRIDE
             ? process.env.VELOCITAS_SDK_PATH_OVERRIDE
-            : join(sdkConfig.getPackageDirectory(), 'latest');
+            : join(coreConfig.getPackageDirectory(), 'latest');
 
         return join(basePath, '.project-creation', 'run.py');
     }
 
     async run(): Promise<void> {
+        let createData: CreateData;
         const packageIndex = PackageIndex.read();
         this._loadDataFromPackageIndex(packageIndex);
+
         const { flags } = await this.parse(Create);
         this.log(`Creating a new Velocitas project ...`);
 
-        if (flags.name && flags.example) {
-            throw new Error("Flags 'name' and 'example' are mutually exclusive!");
-        }
-
-        if (flags.example) {
-            flags.name = flags.example;
-        }
-
         if (Object.keys(flags).length === 0) {
             this.log('Interactive project creation started');
-            await this._runInteractiveMode(flags);
-        }
-
-        if (!flags.name) {
-            throw new Error("Missing required flag 'name'");
-        }
-        if (!flags.language) {
-            throw new Error("Missing required flag 'language'");
-        }
-
-        if (!flags.example && flags.interface) {
-            if (this.appManifestInterfaces.interfaces.length === 0 && flags.interface.length > 0) {
-                await this._handleAdditionalInterfaceArgs(flags.interface);
-            }
+            createData = await this._runInteractiveMode(packageIndex);
+        } else {
+            createData = await this._parseFlags(packageIndex, flags);
         }
-
-        await ProjectConfig.create(packageIndex.getExtensions(), flags.language, this.config.version);
-        await createAppManifest(flags.name, this.appManifestInterfaces);
-        const sdkConfig = new SdkConfig(flags.language);
-        await sdkDownloader(sdkConfig).downloadPackage({ checkVersionOnly: false });
+        await ProjectConfig.create(packageIndex.getMandatoryPackages(), this.config.version);
+        createData.appManifest.write();
+        const coreConfig = new CorePackageConfig(createData.coreId, packageIndex.getPackageByComponentId(createData.coreId).package);
+        await coreDownloader(coreConfig).downloadPackage({ checkVersionOnly: false });
 
         const result = await awaitSpawn(
```

Comment:
I know we discussed something different yesterday, but I guess it would be cleaner if we just call `await Exec.run([createData.coreId, "project-creation"]);` here to leave the exact script path to the core component to define.